app/services/catalog/create_approval_request.rb

```diff
@@ -0,0 +1,62 @@
+module Catalog
+  class CreateApprovalRequest
+    attr_reader :order
+
+    def initialize(id)
+      @order = Order.find_by!(:id => id)
+    end
+
+    def process
+      @order.order_items.each do |order_item|
+        approvals = send_approval_requests(order_item)
+        approvals.compact.each do |req|
+          order_item.approval_requests << create_approval_request(req)
+        end
+      end
+
+      @order.update(:state => "Ordered", :ordered_at => Time.now.utc)
+      self
+    rescue Catalog::ApprovalError => e
+      Rails.logger.error("Error putting in approval Request for #{order.id}: #{e.message}")
+      raise
+    end
+
+    private
+
+    def send_approval_requests(order_item)
```

Comment:
Possibly `submit_approval_requests`? I thinking that makes more sense due to the contents of the log message:
```
Approval Request Submitted for workflow #{workflow}"
```

Reply:
done!